### ISSUE TITLE
Seq.filter on big array fails with maximum call stack size error

### ISSUE THREAD

**Issue by whitetigle** [fable-compiler/Fable#459]
### Description

Encountered problem: on big arrays, Seq.filter may never stops if condition is never met and wll reach maximum call stack threshold.

On my PC, maximum call stack size allowed is for an array with 31327 ints.

### Repro steps

```f#
let testA =
  let max = 1000000
  let a = [| for i in 1 .. max -> 100 |> int  |]
  console.log(a.Length)
  let b = a |> Seq.filter( fun x -> x > 10) |> Seq.toArray
  console.log(b.Length)
  
let testB =
  let max = 1000000
  let a = [| for i in 1 .. max -> (Math.random() * 100.) |> int  |]
  console.log(a.Length)
  let b = a |> Seq.filter( fun x -> x > 10) |> Seq.toArray
  console.log(b.Length)

let testC =
  let max = 1000000
  let a = [| for i in 1 .. max -> 0  |] // init with 0
  console.log(a.Length)
  let b = a |> Seq.filter( fun x -> x > 10) |> Seq.toArray
  console.log(b.Length)

testA // will never fail, condition will be met
testB // may fail if condition is never met
testC // fails with a maximum call stack size error since there aren't any value superior to 10

```
* Fable Version 0.6.5
* Operating system windows 10 / chrome browser

**Comment by alfonsogarciacaro** [comment]
Thanks a lot for the report! This should be fixed in fable-core@0.6.8, could you please give it a try?

**Comment by whitetigle** [comment]
I confirm it works! Bravo 👍
